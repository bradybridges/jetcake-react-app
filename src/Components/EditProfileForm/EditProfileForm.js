import React, { Component } from 'react';
import './EditProfileForm.scss';
import * as firebase from 'firebase';
import 'firebase/storage';
import 'firebase/auth';

export default class EditProfileForm extends Component {
  state = {
    email: this.props.email,
    address: this.props.address,
    phone: this.props.phone,
    dob: this.props.dob,
    securityOne: this.props.securityOne,
    securityTwo: this.props.securityTwo,
    securityThree: this.props.securityThree,
    photoURL: this.props.photoURL,
  };

  handleChange = (e) => {
    this.setState({ [e.target.name]: e.target.value });
  }

  handleUpdateProfile = async () => {
    const { handleUpdateProfile, handleUpdateProfileImg, toggleShowEditProfile } = this.props;
    const stateKeys = Object.keys(this.state);
    stateKeys.forEach((property) => {
      if(this.state[property] !== this.props[property] && property !== "photoURL") {
        const newValue = this.state[property];
        handleUpdateProfile(property, newValue);
      }
      if(property === 'photoURL' && this.state[property] !== this.props[property]) {
        const image = this.state.photoURL;
        handleUpdateProfileImg(image);       
      }
    });
    toggleShowEditProfile(false);
  }

  render() {
    const { email, address, phone, dob, securityOne, securityTwo, securityThree } = this.state;
    const { toggleShowEditProfile } = this.props;
    return (
      <section id="edit-profile-form">
        <label className="input-label">Profile Picture</label>
        <input className="input-field" type="file" onChange={(e) => this.setState({ photoURL: e.target.files[0]})}/>
        <label className="input-label">Email</label>
        <input className="input-field" name="email" type="text" value={email} onChange={this.handleChange} />
        <label className="input-label">Address</label>
        <input className="input-field" name="address" type="text" value={address} onChange={this.handleChange} />
        <label className="input-label">Phone</label>
        <input className="input-field" name="phone" type="text" value={phone} onChange={this.handleChange} />
        <label className="input-label">Birthday</label>
        <input className="input-field" name="dob" type="text" value={dob} onChange={this.handleChange} />
        <label className="input-label">Mother's maiden name?</label>
        <input className="input-field" name="securityOne" type="text" value={securityOne} onChange={this.handleChange} />
        <label className="input-label">Name of first pet?</label>
        <input className="input-field" name="securityTwo" type="text" value={securityTwo} onChange={this.handleChange} />
        <label className="input-label">City you grew up in?</label>
        <input className="input-field" name="securityThree" type="text" value={securityThree} onChange={this.handleChange} />
        <section id="save-profile-btn-container">
          <button className="form-btn" onClick={this.handleUpdateProfile}>Save Profile</button>
          <button className="form-btn" onClick={toggleShowEditProfile}>Cancel</button>
        </section>
      </section>
    );
  }
}
