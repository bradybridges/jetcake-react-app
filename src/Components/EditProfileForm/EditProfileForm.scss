#edit-profile-form {
  display: flex;
  flex-direction: column;
  width: 50%;
  align-items: center;
  background-color: rgba(0, 0, 0, 0.623);
  padding: 10px;
  border-radius: 4px;
}

#save-profile-btn-container {
  margin-top: 10px;
}