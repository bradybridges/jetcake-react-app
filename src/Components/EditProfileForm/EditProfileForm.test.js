import React from 'react';
import { shallow } from 'enzyme';
import EditProfileForm from './EditProfileForm';

describe('EditProfileForm', () => {
  let wrapper;
  const mockHandleUpdateProfile = jest.fn();
  const mockToggleShowEditProfile = jest.fn();

  beforeEach(() => {
    wrapper = shallow(<EditProfileForm
      email="test@test.com"
      address="123 hello ct"
      phone="1112223333"
      dob="1999-01-01"
      securityOne="James"
      securityTwo="Fido"
      securityThree="Atlanta"
      handleUpdateProfile={mockHandleUpdateProfile}
      toggleShowEditProfile={mockToggleShowEditProfile}
    />);
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('handleChange should update the corresponding property in state with a specified value', () => {
    const mockEmailEvent = {
      target: {
        name: 'email',
        value: 'newEmail@newEmail.com',
      },
    };
    const mockPhoneEvent = {
      target: {
        name: 'phone',
        value: '9999999999',
      },
    };
    expect(wrapper.state('email')).toEqual('test@test.com');
    wrapper.instance().handleChange(mockEmailEvent);
    expect(wrapper.state('email')).toEqual('newEmail@newEmail.com');
    wrapper.instance().handleChange(mockPhoneEvent);
    expect(wrapper.state('phone')).toEqual('9999999999');
  });

  it('handleUpdateProfile should call parent function for every property that has changed', async () => {
    wrapper.setState({ email: 'newEmail@yahoo.com', phone: '1234567891' });
    await wrapper.instance().handleUpdateProfile();
    expect(mockHandleUpdateProfile).toHaveBeenCalledWith('email', 'newEmail@yahoo.com');
    expect(mockHandleUpdateProfile).toHaveBeenCalledWith('phone', '1234567891');
  });
});
