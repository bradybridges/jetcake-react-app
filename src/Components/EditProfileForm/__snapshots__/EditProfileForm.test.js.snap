// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EditProfileForm should match snapshot 1`] = `
<section
  id="edit-profile-form"
>
  <label>
    Email
  </label>
  <input
    name="email"
    onChange={[Function]}
    type="text"
    value="test@test.com"
  />
  <label>
    Address
  </label>
  <input
    name="address"
    onChange={[Function]}
    type="text"
    value="123 hello ct"
  />
  <label>
    Phone
  </label>
  <input
    name="phone"
    onChange={[Function]}
    type="text"
    value="1112223333"
  />
  <label>
    Birthday
  </label>
  <input
    name="dob"
    onChange={[Function]}
    type="text"
    value="1999-01-01"
  />
  <label>
    Mother's maiden name?
  </label>
  <input
    name="securityOne"
    onChange={[Function]}
    type="text"
    value="James"
  />
  <label>
    Name of first pet?
  </label>
  <input
    name="securityTwo"
    onChange={[Function]}
    type="text"
    value="Fido"
  />
  <label>
    City you grew up in?
  </label>
  <input
    name="securityThree"
    onChange={[Function]}
    type="text"
    value="Atlanta"
  />
  <button
    onClick={[Function]}
  >
    Save Profile
  </button>
  <button
    onClick={[MockFunction]}
  >
    Cancel
  </button>
</section>
`;
